src/c20_client/do_client_job.py

```diff
@@ -0,0 +1,51 @@
+import argparse
+import requests
+
+from c20_client.client_logger import LOGGER
+
+from c20_client.connection_error import NoConnectionError
+from c20_client.client import get_result_for_job
+from c20_client.do_wait import wait_between_jobs
+
+
+def do_job(api_key):
+    """
+    Gets a job from the server and handles the job based on the type of job
+    """
+    try:
+        LOGGER.info('Getting job from server...')
+        job = requests.get('http://capstone.cs.moravian.edu/get_job')
+        job = job.json()
+        LOGGER.info("Job acquired")
+
+    except Exception:
+        raise NoConnectionError
+
+    get_result_for_job(job, api_key)
+
+
+def do_multiple_job(api_key):
+    """
+       Gets multiple job from the server
+       and handles the job based on the type of job
+    """
+    try:
+        print("Getting job...\n")
+        while True:
+            do_job(api_key)
+            wait_between_jobs()
```

Comment:
This is the untestable code because it has the `while(True)` loop.  Therefore, the call to `time.sleep` should be here as well (it is also problematic to test).

```
while True:
   do_job(api_key)
   time.sleep(get_wait_time())
```

This also resolves the multi-layer functions I mentioned in another comment.

Reply:
This version will work for the happy path (jobs successful), but we will need more for handling errors.

Something like:

```
while True:
    status = do_job(api_key)
    time.sleep(get_wait_time(status))
```

Let's get the happy path merged in before we tackle this...